local.c

```diff
@@ -117,6 +122,10 @@ static void local_shutdown(struct iio_context *ctx)
 			close(dev->pdata->fd);
 		local_free_pdata(dev);
 	}
+
+	if (ctx->pdata && ctx->pdata->lock)
```

Comment:
do we really need the condition?

Reply:
Yes, because local_create_context() can fail before the pdata is allocated or the mutex is created. The error path calls iio_context_destroy() which in turn calls local_shutdown().